### DISCUSSION TITLE
Tasmota on WT32-ETH01 without wifi and BT to free up io pins for ADC

### DISCUSSION THREAD

**Discussion by ON4BCB** [arendst/Tasmota#22484] [Ideas]
rather new to tasmota, so thanks for pointing me to the right answer if it excists, i red alot but not everything

First thanks to Arendst for tasmota

I would like to use the WT32-ETH01 hardware and use all possible ADC's
I notice in the latest version that even if wifi is not configed, IO/ADC 14 and 15 are not working
Is therea way to use the WT32-ETH01 with ethernet only ?

Many thanks

**Comment by sfromis** [comment]
Tasmota does not support ADC2, only ADC1. While ADC2 is "problematic" with Wifi, it does not change things by turning off `Wifi` (that't the command name).

**Comment by ON4BCB** [comment]
it seems that

Configure the Device to Use Ethernet Only

   1.

   *Set the device to Ethernet mode* by going to the *Console* in Tasmota
   (found in the web interface) and typing the following command:

   plaintext
   Copy code
   Ethernet 1

   This command enables the Ethernet connection as the primary
   communication method.
   2.

   *Disable Wi-Fi*:
   - To turn off Wi-Fi and prevent the device from connecting to any Wi-Fi
      network, enter the following command:

      plaintext
      Copy code
      Wifi 0

      - Setting Wifi 0 will disable the Wi-Fi entirely. Alternatively, Wifi
      1 would enable it again if needed.
   3.

   *Save and Restart*:
   - After setting these parameters, you may need to restart the device to
      apply the changes. Use the command:

      plaintext
      Copy code
      Restart 1

      4. at least is giving ADC reads in the main webpage, seems to work
   unless i'm wrong



Op do 14 nov 2024 om 15:16 schreef sfromis ***@***.***>:

**Reply by sfromis** [reply]
Until you've verified getting varying readings, you've confirmed nothing.

From the Tasmota source code:
`const uint8_t MAX_ADCS = 8; // Max number of ESP32 ADC pins (ADC2 pins are unusable with Wifi enabled)`

**Comment by Jason2866** [comment]
Do some investigation and search in espressif IDF documentation. In short switching off wifi will not work to get usable results for ADC2. Reason the Arduino libs are compiled with wifi enabled.
